README.md

```diff
@@ -1,23 +1,70 @@
 
 ## Prerequisites
 
 Prepare your ruleset for bzlmod by following the [Bzlmod User Guide](https://bazel.build/docs/bzlmod).
 
-## How it works
+Then, include these [template files](./templates) in your ruleset repository.
 
-1. [Configure](https://github.com/apps/publish-to-bcr) the app for two repositories:
+## Setup
 
-   - Your ruleset repository.
-   - A fork of [bazelbuild/bazel-central-registry](https://github.com/bazelbuild/bazel-central-registry). The fork can be in the same GitHub account as your ruleset _or_ in the release author's personal account. If you use release automation and the release author is the github-actions bot, then the fork must
-     be in ruleset's account unless you [override the releaser](./templates/README.md#optional-configyml).
+Create a GitHub Actions workflow in your ruleset repository by creating a file, typically named `.github/workflows/publish.yaml`.
 
-   _Note: Authors of rulesets under the `bazelbuild` org should add the app to their personal fork of `bazelbuild/bazel-central-registry`._
+This repository provides a reusable workflow that contains all the boilerplate.
+See complete documentation in the [reusable workflow file](./.github/workflows/publish.yaml).
+
+1. Decide how your workflow will be invoked
+
+Use an `on` block, and provide at least the `tag_name` as an input.
+
+For example, if you have a release automation using GitHub Actions, you might call this publish workflow upon successful completion.
+A recommended setup for release automation, including generating attestations to prove the provenance of release artifacts, may be found at
+https://github.com/bazel-contrib/.github/blob/master/.github/workflows/
+
+As another example, you might use `workflow_dispatch` to manually run the publish workflow from the GitHub web UI or the CLI
+([documentation](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/manually-running-a-workflow))
+
+It's also useful to permit both of these, for example:
+
+```yaml
+on:
+  # Run the publish workflow after a successful release
+  # Will be triggered from the release.yaml workflow
+  workflow_call:
+    inputs:
+      tag_name:
+        required: true
+        type: string
+  # In case of problems, let release engineers retry by manually dispatching
+  # the workflow from the GitHub UI
+  workflow_dispatch:
+    inputs:
+      tag_name:
+        required: true
+        type: string
+```
+
+2. Reference the reusable workflow in your `publish` job (replacing the `[version]` placeholder).
+
+```yaml
+jobs:    
+  publish:
+    uses: bazel-contrib/publish-to-bcr/.github/workflows/publish.yaml@[version]
+    with:
+      tag_name: ${{ inputs.tag_name }}
+      # GitHub repository which is a fork of the upstream where the Pull Request will be opened.
+      registry_fork: aspect-build/bazel-central-registry
+    permissions:
+      contents: write  # allow creating a PR
```

Comment:
I don't think this is correct. The PUBLISH_TOKEN is what allows pushing to the fork and opening the pull request. `contents: write` is necessary because the publish workflow uploads attestations to the release.

Reply:
doesn't the release workflow put the attestations on the release? I don't see anywhere in the publish workflow that seems to need write

Reply:
The release workflow uploads attestation for the release archive(s). The publish workflow uploads attestations for source.json and MODULE.bazel. So it also needs the attestation permissions.